Userland/Libraries/LibC/netinet/ip6.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2024, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <assert.h>
+#include <inttypes.h>
+#include <netinet/in.h>
+
+__BEGIN_DECLS
+
+struct ip6_hdr {
+    union {
+        struct ip6_hdrctl {
+            uint32_t ip6_un1_flow; // 4 bits version, 8 bits TC, 20 bits flow-ID
+            uint16_t ip6_un1_plen; // payload length
+            uint8_t ip6_un1_nxt;   // next header
+            uint8_t ip6_un1_hlim;  // hop limit
+        } ip6_un1;
+        uint8_t ip6_un2_vfc; // 4 bits version, top 4 bits tclass
+    } ip6_ctlun;
+    struct in6_addr ip6_src; // source address
+    struct in6_addr ip6_dst; // destination address
+} __attribute__((packed));
+static_assert(sizeof(struct ip6_hdr) == 40, "struct ip6_hdr: invalid length");
```

Comment:
We don’t usually do this in C headers.

Reply:
It was used in [ip.h](https://github.com/SerenityOS/serenity/blob/0482f4e1170010dccec1726ee5222a062f4b3262/Userland/Libraries/LibC/netinet/ip.h#L35)